sdk/security/keyvault/azkeys/constants.go

```diff
@@ -9,244 +9,191 @@
 
 package azkeys
 
-// DeletionRecoveryLevel - Reflects the deletion recovery level currently in effect for keys in the current vault. If it contains
-// 'Purgeable' the key can be permanently deleted by a privileged user; otherwise, only the system
-// can purge the key, at the end of the retention interval.
-type DeletionRecoveryLevel string
+// KeyCurveName - Elliptic curve name. For valid values, see JsonWebKeyCurveName.
+type KeyCurveName string
```

Comment:
How about just `CurveName`?

Reply:
I prefer `KeyCurveName` since it matches the other languages. This type is also used in `azcertificates` and the Key prefix adds clarity on what the type is for.

Reply:
Matching other languages isn't important because developers read our docs when writing Go. I could live with `KeyCurveName`, but I do think `CurveName` is better, and precise in the context of crypto keys. (If you're really worried about a vague name here, `EllipticCurveName` removes all doubt.)

I don't think the "Key" prefix helps developers figure out how to use this type. Developers don't list a package's types and try to intuit by their names how to use them. They start with method calls, which lead them to the relevant models, whose fields lead them to the relevant types, whose names aren't so important because the compiler points out the right one.

Reply:
Right, the "Key" prefix in azcertificates was to indicate it was the curve name of the certificate backing key. It was significant there, but seemingly redundant here in azkeys.

Reply:
Will keep as `CurveName`. Will also have it as `CurveName` in `azcertificates` for consistency.